### PR TITLE
luminous: tests: qa/standalone: misc fixes

### PR DESCRIPTION
**Pull request by dzafman**
Backport: http://tracker.ceph.com/issues/20979
including https://github.com/ceph/ceph/pull/17026
including https://github.com/ceph/ceph/pull/17093
including https://github.com/ceph/ceph/pull/17098

### CODE REVIEW

**Comment by dzafman** [comment]
I want to add a cherry-pick of #17026 to this pull request.

File: qa/run-standalone.sh [hunk 1]
```diff
@@ -1,25 +1,116 @@
...
+    KERNCORE="kernel.core_pattern"
+    COREPATTERN="core.%e.%p.%t"
+fi
+
+function finish() {
+    if [ -n "$precore" ]; then
+        sudo sysctl -w ${KERNCORE}=${precore}
+    fi
+    exit 0
+}
```
Comment (wjwithagen) [inline qa/run-standalone.sh]: @dzafman 
Exiting here with `0` will return `OKE` which is probably no the correct information once you run this in a testing environment.

**Comment by theanalyst** [comment]
@dzafman are the OSD related changes needed in this patchset itself, and does this need to go in for 12.2.0?

**Comment by dzafman** [comment]
@theanalyst The osd and test changes fix all run-standalone.sh tests to run without failure.  They rarely fail in official runs of teuthology because a subset run won't run many of these tests.  I wouldn't have a problem if you wanted to exclude it, but it is ready to go.

**Comment by dzafman** [comment]
@liewegas You approved this, but do you want it in now or the next RC or point release?

**Comment by liewegas** [comment]
http://pulpito.ceph.com/sage-2017-08-23_20:25:11-rados-wip-sage-testing-luminous-20170823a-distro-basic-smithi/

**Pull request merged**